Allow to run tests on Linux

ikhaldeev opened the pull request:
Allow to run tests and therefore build on Linux platform.

A platform is determined by "os.name" system property.

rillig commented on java/test/com/google/typography/font/sfntly/FontTest.java:
```diff
@@ -87,13 +87,30 @@ public void testReadInvalidFile() throws IOException {
+      fontsDir = new File("C:/Windows/Fonts");
+      Assume.assumeTrue(fontsDir.exists());
+    } else if (isLinux()) {
+      fontsDir = new File("/usr/share/fonts");
```
I like the idea. But did you actually test this patch? On my freshly installed Ubuntu Xenial, there are no `*.ttf` files directly in `/usr/share/fonts`. They are all in some subdirectories, but those are currently not covered by this test.

rillig commented on java/test/com/google/typography/font/sfntly/FontTest.java:
```diff
@@ -87,13 +87,30 @@ public void testReadInvalidFile() throws IOException {
+    File fontsDir;
+    if (isWindows()) {
+      fontsDir = new File("C:/Windows/Fonts");
+      Assume.assumeTrue(fontsDir.exists());
```
This assumption should be outside of the `if` since it applies to each and all environments.

ikhaldeev commented:
@rillig Thanks for your comments!
I've tested that it does not fail, but you are right. I've double checked it and it appears that it actually tests nothing, as it could not find any font and it never invokes TestFontUtils.loadFont. 
I've changed the iteration so it checks folders recursively. Also, I've added an assertion that at least one font is loaded.

ikhaldeev commented:
btw, I've checked this only on Ubuntu.

rillig commented:
Thanks for the patch.
